src/network/networkd-address.c

```diff
@@ -448,6 +448,127 @@ static int address_set_masquerade(Address *address, bool add) {
         return 0;
 }
 
+static int netlabel_op(bool add, const char *label, const Address *address) {
+        _cleanup_(sd_netlink_unrefp) sd_netlink *genl = NULL;
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *m = NULL;
+        int r;
+
+        assert(address);
+        assert(address->link);
+        assert(address->link->network);
+
+        r = sd_genl_socket_open(&genl);
+        if (r < 0)
+                return r;
+
+        uint16_t command;
+        if (add)
+                command = NLBL_UNLABEL_C_STATICADD;
+        else
+                command = NLBL_UNLABEL_C_STATICREMOVE;
+
+        r = sd_genl_message_new(genl, NETLBL_NLTYPE_UNLABELED_NAME, command, &m);
+        if (r < 0)
+                return r;
+
+        r = sd_netlink_message_append_string(m, NLBL_UNLABEL_A_IFACE, address->link->ifname);
+        if (r < 0)
+                return r;
+
+        if (add) {
+                r = sd_netlink_message_append_string(m, NLBL_UNLABEL_A_SECCTX, label);
+                if (r < 0)
+                        return r;
+        }
+
+        union in_addr_union mask;
+
+        memset(&mask, 0xff, sizeof(mask));
+        r = in_addr_mask(address->family, &mask, address->prefixlen);
```

Comment:
Why do you initialize the mask with `0xff` here?? What value should we pass to the kernel?

Reply:
I think the mask should have bits set according to prefixlen, like 255.255.0.0 for prefix 16 in IPv4. Perhaps it would be better instead to add a new function `in_addr_prefixlen_to_netmask()`, using existing `in4_addr_prefixlen_to_netmask()` and (not yet existing) `in6_addr_prefixlen_to_netmask()`.